business_rules/operators.py

```diff
@@ -938,6 +939,68 @@ def is_ordered_by(self, other_value: dict) -> pd.Series:
     def is_not_ordered_by(self, other_value: dict) -> pd.Series:
         return ~self.is_ordered_by(other_value)
 
+    @type_operator(FIELD_DATAFRAME)
+    def target_is_sorted_by(self, other_value: dict) -> pd.Series:
+        """
+         Checking if  SESEQ is  sorted based on SESTDTC within USUBJID
+        """
+
+        target: str = self.replace_prefix(other_value.get("target"))
+        columns = list(other_value["user_defined_columns"].keys())
+
+        for index in range(0, len(columns), 2):
+            within: str = self.replace_prefix(columns[index])
+            comparator: str = self.replace_prefix(columns[index+1])
+
+            temp_seseq = self.create_count_map(within)
+
+            if type(self.value[within][0]) is str:
```

Comment:
What are these type checks for? if the within is not a variable name, an error that the user can understand will be thrown

Reply:
These checks are for test cases when the columns are ASCII and not numerical.

Reply:
For a dataset consisting of alphabets like 
{
                "USUBJID": ['a', 'b', 'b', 'a', 'a'],
                "SESEQ": ['a', 'b', 'a', 'c', 'b'],
                "SESTDTC": ['g', 'i', 'f', 'j', 'h'],
}
SESEQ has alphabetical counts that's why to convert the cumulative count to its corresponding alphabetical letter we are using this check.

Reply:
Can you show values of temp_seseq before the operations and after?

Reply:
before 
temp_seseq =[1,1,2,2,3]
after the operation 
temp_seseq =['a', 'a', 'b' ,'b' ,'c']

Reply:
Why are you doing that? We should not change the data type. We can sort numbers and characters the same way